src/main.c

```diff
@@ -79,6 +80,11 @@ int main(void)
 
     set_green_led(false);
     set_red_led(true);
+	
+	if (pulse_counter_enabled){
+		log_info("Pulse Counter Init");
+		pulse_counter_init();
+	}
 
     if (gps_nmea_output_enabled) {
```

Comment:
Please note that you'll have to prevent initialization (and further usage) of both the USART and I²C in case pulse counter is enabled. Enabling I²C/USART/pulse counter should all be mutually exclusive.

Reply:
I think `i2c_init()` is now reconfiguring the pin you're trying to use and that's why GPIO interrupts are not working.

Reply:
@ManoDaSilva

Reply:
@mikaelnousiainen That makes total sense! I added some interlocks (see 2 following commits), but still no interrupt... I even tried removing I2C/UART init altogether, but same issue... Probably something wrong with my GPIO/Interrupts config?

Reply:
Second mistake: forgot GPIO_EXTILineConfig(GPIO_PortSourceGPIOB, GPIO_PinSource11);, which links the interrupt line to the GPIO port. Now works on a bare-bones code (with an LED), off to integrate it into this and check if it doesn't interfere with anything else...